### ISSUE TITLE
Email provider fails

### ISSUE THREAD

**Issue by AL1L** [nextauthjs/next-auth#1717]
**Describe the bug**
When using the email provider, it fails because `provider.protection` is undefined, see error below

**Steps to reproduce**

1. Create project with `yarn create next-app` or `npx create-next-app`, add next-auth using getting started guide, add email provider like so

  ```js
  import NextAuth from "next-auth";
  import Providers from "next-auth/providers";
  
  export default NextAuth({
      providers: [
          Providers.Email({
  
              server: {
                  host: process.env.EMAIL_SERVER_HOST,
                  port: process.env.EMAIL_SERVER_PORT,
                  auth: {
                      user: process.env.EMAIL_SERVER_USER,
                      pass: process.env.EMAIL_SERVER_PASSWORD
                  },
              },
              from: process.env.EMAIL_FROM,
          }),
      ],
  
      // mongodb://username:password@host:port/database_name
      database: process.env.DATABASE_URL,
  });
  ```
  The same also happens when using a SMTP URI.

2. Click signin button
3. enter email
4. notice it takes you to http://localhost:3000/api/auth/signin?error=OAuthSignin and there is now an error in the Nodejs console.

(do not feel like creating an example repo, you guys can do that if you want.)

**Expected behavior**
Email login should work when putting in valid email and email servers.

**Screenshots or error logs**
```[next-auth][error][signin_oauth_error] 
https://next-auth.js.org/errors#signin_oauth_error TypeError: Cannot read property 'includes' of undefined
    at Object.<anonymous> (C:\Users\AL_1\Projects\Artex\studio-archetype\node_modules\next-auth\dist\server\lib\oauth\pkce-handler.js:101:32)
    at Generator.next (<anonymous>)
    at asyncGeneratorStep (C:\Users\AL_1\Projects\Artex\studio-archetype\node_modules\next-auth\dist\server\lib\oauth\pkce-handler.js:32:103)
    at _next (C:\Users\AL_1\Projects\Artex\studio-archetype\node_modules\next-auth\dist\server\lib\oauth\pkce-handler.js:34:194)
    at C:\Users\AL_1\Projects\Artex\studio-archetype\node_modules\next-auth\dist\server\lib\oauth\pkce-handler.js:34:364
    at new Promise (<anonymous>)
    at Object.<anonymous> (C:\Users\AL_1\Projects\Artex\studio-archetype\node_modules\next-auth\dist\server\lib\oauth\pkce-handler.js:34:97)
    at Object.handleSignin (C:\Users\AL_1\Projects\Artex\studio-archetype\node_modules\next-auth\dist\server\lib\oauth\pkce-handler.js:88:24)
    at C:\Users\AL_1\Projects\Artex\studio-archetype\node_modules\next-auth\dist\server\index.js:236:32
    at Generator.next (<anonymous>)
    at asyncGeneratorStep (C:\Users\AL_1\Projects\Artex\studio-archetype\node_modules\next-auth\dist\server\index.js:52:103)
    at _next (C:\Users\AL_1\Projects\Artex\studio-archetype\node_modules\next-auth\dist\server\index.js:54:194)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
```

**Additional context**
Adding `protection: 'none',` to the Email provider options makes it work.

Here are my package.json dependencies 
```json
{
  "dependencies": {
    "mongodb": "^3.6.6",
    "mongoose": "^5.12.3",
    "next": "10.1.3",
    "next-auth": "^3.14.0",
    "react": "17.0.2",
    "react-dom": "17.0.2",
    "sass": "^1.32.8"
  }
}
```

**Feedback**
*Documentation refers to searching through [online documentation](https://next-auth.js.org), code comments and issue history. The example project refers to [next-auth-example](https://github.com/iaincollins/next-auth-example).*

* [x] Found the documentation helpful
* [ ] Found documentation but was incomplete
* [ ] Could not find relevant documentation
* [ ] Found the example project helpful
* [ ] Did not find the example project helpful

**Comment by AL1L** [comment]
Looks like this was fixed in #1694

**Comment by AL1L** [comment]
And is a dupe of #1692 lol
